datafusion/spark/src/function/bitmap/bitmap_count.rs

```diff
@@ -100,6 +101,22 @@ pub fn bitmap_count_inner(arg: &[ArrayRef]) -> Result<ArrayRef> {
         FixedSizeBinary(_size) => {
             downcast_and_count_ones!(input_array, FixedSizeBinaryArray)
         }
+        Dictionary(k, v) if k.as_ref() == &DataType::Int32 && v.as_ref() == &Binary => {
+            let dict_array = as_dictionary_array::<Int32Type>(input_array);
+            let binary_array = as_binary_array(dict_array.values())?;
+
+            let result: Int64Array = dict_array
+                .keys()
+                .iter()
+                .map(|key| {
+                    key.and_then(|k| {
+                        binary_count_ones(Some(binary_array.value(k as usize)))
+                    })
+                })
+                .collect();
+
+            Ok(result)
+        }
```

Comment:
```suggestion
        Dictionary(k, v) if k.as_ref() == &DataType::Int32 && v.as_ref() == &Binary => {
            let dict_array = as_dictionary_array::<Int32Type>(input_array);
            let array = dict_array.downcast_dict::<BinaryArray>().unwrap();
            Ok(array
                .into_iter()
                .map(binary_count_ones)
                .collect::<Int64Array>())
        }
```

Using [`TypedDictionaryArray`](https://docs.rs/arrow/latest/arrow/array/struct.TypedDictionaryArray.html)

Reply:
Thanks, fixed.